discovery-infra/test_infra/helper_classes/kube_helpers.py

```diff
@@ -0,0 +1,376 @@
+            'spec': {
+                'clusterName': self.ref['name'],
+                'baseDomain': base_domain,
+                'platform': platform,
+                'provisioning': {'installStrategy': install_strategy},
+                'pullSecretRef': secret.ref,
+            }
+        }
+        body['spec'].update(kwargs)
+        self.crd_api.create_namespaced_custom_object(
+            group=self.hive_api_group,
+            version='v1',
+            plural='clusterdeployments',
+            body=body,
+            namespace=self.ref['namespace']
+        )
+        self.__secret = secret
+
+        logger.info('created cluster deployment %s: %s', self.ref, body)
+
+    def get(self) -> dict:
+        return self.crd_api.get_namespaced_custom_object(
+            group=self.hive_api_group,
+            version='v1',
+            plural='clusterdeployments',
+            name=self.ref['name'],
+            namespace=self.ref['namespace']
+        )
+
+    def delete(self) -> None:
+        self.crd_api.delete_namespaced_custom_object(
+            group=self.hive_api_group,
+            version='v1',
+            plural='clusterdeployments',
+            name=self.ref['name'],
+            namespace=self.ref['namespace']
+        )
+
+        logger.info('deleted cluster deployment %s', self.ref)
+
+    def status(
+            self,
+            timeout: Union[int, float] = DEFAULT_WAIT_FOR_CRD_STATUS_TIMEOUT
+    ) -> dict:
+
+        # 'status' key is created only after resource is processed by the
+        # controller in assisted-service. It might take a few seconds before
+        # appears.
+        def _attempt_to_get_status() -> dict:
+            return self.get()['status']
+
+        return waiting.wait(
+            _attempt_to_get_status,
+            sleep_seconds=0.5,
+            timeout_seconds=timeout,
+            waiting_for=f'cluster {self.ref} status',
+            expected_exceptions=KeyError
+        )
+
+    def state(
+            self,
+            timeout: Union[int, float] = DEFAULT_WAIT_FOR_CRD_STATE_TIMEOUT
+    ) -> Tuple[str, str]:
+
+        state, state_info = None, None
+        for condition in self.status(timeout).get('conditions', []):
+            reason = condition.get('reason')
+
+            if reason == 'AgentPlatformState':
+                state = condition.get('message')
+
+            # todo: fix typo after it will be fixed in assisted-service
+            #  (Agnet -> Agent)
+            elif reason == 'AgnetPlatformStateInfo':
+                state_info = condition.get('message')
+
+            if state and state_info:
+                break
+
+        return state, state_info
+
+    def wait_for_state(
+            self,
+            required_state: str,
+            timeout: Union[int, float] = DEFAULT_WAIT_FOR_CRD_STATE_TIMEOUT
+    ) -> None:
+
+        required_state = required_state.lower()
+
+        def _has_required_state() -> bool:
+            state, _ = self.state(0.5)
```

Comment:
Shouldn't it use `timeout` parameter?

Reply:
nope, the timeout is for the whole waiting process and not a single status call

Reply:
technically it should have but I'm not counting on it

Reply:
OK. At least make it a constant variable.